src/main/java/com/fakng/fakngagrgtr/parser/stripe/StripeParser.java

```diff
@@ -0,0 +1,150 @@
+
+        for (int i = 0; i < urls.size(); i++) {
+            String jobId = urls.get(i).getAttributes().getNamedItem("href").getNodeValue();
+
+            // Skip repeating vacancies and parse locations for them
+            List<LocationDTO> locationDTOs = new ArrayList<>();
+            while (jobId.equals(urls.get(i).getAttributes().getNamedItem("href").getNodeValue())) {
+                locationDTOs.add(createLocationDto(cities.get(i)));
+                i++;
+                if (i >= urls.size()) break;
+            }
+
+            // Variable i stopped in incorrect position
+            i--;
+            vacancies.add(createVacancy(jobId, locationDTOs));
+        }
+
+        return vacancies;
+    }
+
+    private Vacancy createVacancy(String jobId, List<LocationDTO> locationDTOS) throws IOException {
+        Vacancy vacancy = new Vacancy();
+
+        HtmlPage page = downloadPage(STRIPE_URL + jobId);
+        DomElement title = page.getFirstByXPath(VACANCY_TITLE_XPATH);
+        DomElement jobType = ((DomElement) page.getFirstByXPath(VACANCY_JOB_TYPE_XPATH)).getNextElementSibling();
+        DomElement team = ((DomElement) page.getFirstByXPath(VACANCY_TEAM_XPATH)).getNextElementSibling();
+
+        vacancy.setTitle(title.getTextContent());
+        vacancy.setDescription(createFullDescription(
+                createDescription(page), jobType.getTextContent(), team.getTextContent())
+        );
+        vacancy.setUrl(STRIPE_URL + jobId);
+        vacancy.setJobId(jobId.replaceAll("/.*/", ""));
+        vacancy.setCompany(company);
+        vacancy.setLocations(processLocations(locationDTOS));
+
+        return vacancy;
+    }
+
+    private List<Location> processLocations(List<LocationDTO> locationDTOS) {
+        List<Location> locations = new ArrayList<>();
+        locationDTOS.forEach(locationDTO -> locations.add(locationProcessor.processLocation(
+                    company, locationDTO.getCity(), locationDTO.getCountryCode()
+            ))
+        );
+
+        return locations;
+    }
+
+    private String createDescription(HtmlPage page) {
+        List<DomNode> description = page.getByXPath(VACANCY_DESCRIPTION_BLOCK_XPATH);
+
+        StringBuilder stringBuilder = new StringBuilder();
+        description.forEach(
+                tag -> stringBuilder.append(tag.asNormalizedText()).append("\n")
+        );
+
+        return stringBuilder.toString();
+    }
+
+    private String createFullDescription(String description, String jobType, String team) {
+        return description +
+                "Job Type: " + jobType + "\n" +
+                "Team: " + team;
+    }
+
+    private LocationDTO createLocationDto(DomElement city) {
+        LocationDTO locationDTO = new LocationDTO();
+        locationDTO.setCity(city.getTextContent());
+        // Get country code from alt attribute from the image next to city name
+        locationDTO.setCountryCode(city.getPreviousElementSibling().getAttribute("alt"));
```

Comment:
If you have comments in your code describing what this code does, it basically says that the code is written in a not really good way. Your code should tell for itself without comments.